src/apify/_utils.py

```diff
@@ -410,3 +412,118 @@ def validate_single(field_value: Any, expected_type: type, required: bool, name:
 PARSE_DATE_FIELDS_MAX_DEPTH = 3
 PARSE_DATE_FIELDS_KEY_SUFFIX = 'At'
 ListOrDictOrAny = TypeVar('ListOrDictOrAny', list, dict, Any)
+
+
+def get_short_base64_hash(data: bytes) -> str:
+    """Generates an 8-character, base64-encoded SHA-256 hash of the provided data.
+
+    Args:
+        data: The binary data to be hashed.
+
+    Returns:
+        An 8-character, base64-encoded hash.
+    """
+    # Create a SHA-256 hash of the payload.
+    hash_object = sha256(data)
+    # Convert the hash to a base64-encoded string.
+    base64_encoded_hash = b64encode(hash_object.digest()).decode()
+    # Remove any '+' or '/' or '=' characters and return the first 8 characters.
```

Comment:
I'm just wondering why do you need first 8 characters of a base64 encoded sha256 hash without `+`, `/`, and `=`? Seems rather complicated

Reply:
In this case, I just followed the implementation in the Crawlee - [_hashPayload](https://github.com/apify/crawlee/blob/v3.8.1/packages/core/src/request.ts#L397:L405). We had a short Slack discussion with the team keeping it 1:1 with Crawlee.

```
$ g blame -L 397,+8 packages/core/src/request.ts
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 397)     protected _hashPayload(payload: BinaryLike): string {
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 398)         return crypto
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 399)             .createHash('sha256')
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 400)             .update(payload)
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 401)             .digest('base64')
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 402)             .replace(/[+/=]/g, '')
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 403)             .substring(0, 8);
3a59bfc0b (Martin Adámek 2022-07-13 20:00:07 +0200 404)     }
```

I believe the intention behind this was only to keep alphanumeric characters in the hash. @B4nan Maybe you could clarify it?

Reply:
Hmm, ok, fair enough.
For the alphanumeric representation, hash can be converted to a string representation of a number with base 16, e.g.:

```
const crypto = require('crypto');

const hash = crypto.createHash('sha256');

hash.update('123');
console.log(hash.digest(456).toString('hex'));
```

Gives:
`a665a45920422f9d417e4867efdc4fb8a04a1f3fff1fa07e998e86f7f7a27ae3`, you don't need to do base64 and replace of `+/=`.

But yeah, let's keep the parity

Reply:
IMO this is internal, so we can go this way in this case (if the intention was really to keep only alphanumeric chars).

Reply:
implemented